src/lib.rs

```diff
@@ -67,12 +67,37 @@ mod tests {
         }
     }
 
+    #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+    pub struct NonZeroTinyInt(u8);
```

Comment:
maybe use [`NonZeroU8`](https://doc.rust-lang.org/std/num/struct.NonZeroU8.html) ?

Reply:
:+1: Ok pushed up a change that makes use of the NonZeroU8 type, I think it's better now